src/gui/transferlistwidget.cpp

```diff
@@ -548,16 +552,27 @@ void TransferListWidget::hidePriorityColumn(bool hide)
 void TransferListWidget::openSelectedTorrentsFolder() const
 {
     QSet<QString> pathsList;
+#ifdef Q_OS_MAC
+    // On macOS you expect both the files and folders to be opened in their parent
+    // folders prehilighted for opening, so we use a custom method.
+    foreach (BitTorrent::TorrentHandle *const torrent, getSelectedTorrents()) {
+        QString path = torrent->contentPath(true);
+        if (!pathsList.contains(path))
```

Comment:
No point in check then insert, just insert all the way.
`pathsList.insert(torrent->contentPath(true));`

Reply:
Oops, some messed up copy-pasting from the original one. Fixed now.